### PR TITLE
Swap limit and usage must be 0 in absence of swap limit capabilities

### PR DESCRIPTION
**Pull request by ashu-mehra**
It is possible in certain configurations that cgroup swap limit
capabilities is not enabled. In such cases files in cgroup memory
subsystem related to swap information such as
'memory.memsw.limit_in_bytes' and 'memory.memsw.usage_in_bytes'
won't be present.
Updated retrieveLinuxCgroupMemoryStats() to handle such case.

Signed-off-by: Ashutosh Mehra <asmehra1@in.ibm.com>

### CODE REVIEW

**Comment by ashu-mehra** [comment]
fyi - @mpirvu

**Comment by charliegracie** [comment]
@genie-omr build all

File: port/unix/omrsysinfo.c [hunk 1]
```diff
@@ -1488,10 +1488,27 @@ retrieveLinuxCgroupMemoryStats(struct OMRPortLibrary *portLibrary, struct OMRCgr
 	}
 	rc = readCgroupSubsystemFile(portLibrary, OMR_CGROUP_SUBSYSTEM_MEMORY, CGROUP_MEMORY_SWAP_LIMIT_IN_BYTES_FILE, numItemsToRead, "%lu", &cgroupMemInfo->memoryAndSwapLimit);
 	if (0 != rc) {
-		goto _exit;
+		if (OMRPORT_ERROR_FILE_NOENT == rc) {
+			/* It is possible file memory.memsw.limit_in_bytes is not present if
+			 * swap space is not configured. In such cases, set memoryAndSwapLimit to same as memoryLimit.
+			 */
+			cgroupMemInfo->memoryAndSwapLimit = cgroupMemInfo->memoryLimit;
```
Comment (charliegracie) [inline port/unix/omrsysinfo.c]: what is cgroupMemInfo->memoryAndSwapLimit supposed to represent?

File: port/unix/omrsysinfo.c [hunk 2]
```diff
@@ -1488,10 +1488,27 @@ retrieveLinuxCgroupMemoryStats(struct OMRPortLibrary *portLibrary, struct OMRCgr
...
+			goto _exit;
+		}
 	}
 	rc = readCgroupSubsystemFile(portLibrary, OMR_CGROUP_SUBSYSTEM_MEMORY, CGROUP_MEMORY_SWAP_USAGE_IN_BYTES_FILE, numItemsToRead, "%lu", &cgroupMemInfo->memoryAndSwapUsage);
 	if (0 != rc) {
+		if (OMRPORT_ERROR_FILE_NOENT == rc) {
+			/* It is possible file memory.memsw.usage_in_bytes is not present if
+			 * swap space is not configured. In such cases, set memoryAndSwapUsage to memoryUsage.
+			 */
+			cgroupMemInfo->memoryAndSwapUsage = cgroupMemInfo->memoryUsage;
```
Comment (charliegracie) [inline port/unix/omrsysinfo.c]: what is cgroupMemInfo->memoryAndSwapUsage supposed to represent?

Comment (ashu-mehra) [reply port/unix/omrsysinfo.c]: It represents sum of memory and swap usage. From `port/unix_include/omrcgroup.h`:
```
        uint64_t memoryAndSwapUsage; /**< current memory + swap usage in bytes (as in memory.memsw.usage_in_bytes file) */
```

File: port/unix/omrsysinfo.c (hunk 1 above)
Comment (ashu-mehra) [reply port/unix/omrsysinfo.c]: It represents the sum of memory and swap limit. From `port/unix_include/omrcgroup.h`:
```
        uint64_t memoryAndSwapLimit; /**< memory + swap limit in bytes (as in memory.memsw.limit_in_bytes file)*/
```

**Comment by charliegracie** [comment]
Merging without waiting for AppVeyor since the only file changed is not used on Windows

**Pull request merged**